File: src/IsbInterop/ServiceForms/Proxies/EdmsExplorer.vb
﻿Imports IsbInterop.Accessory
Imports IsbInterop.Accessory.Proxies
Imports IsbInterop.DataTypes.Enumerable
Imports IsbInterop.Folders

Namespace ServiceForms.Proxies

  ''' <summary>
  ''' Обертка над IEdmsExplorer.
  ''' </summary>
  Public Class EdmsExplorer
    Inherits BaseIsbObject
    Implements IEdmsExplorer

    ''' <summary>
    ''' Тип отображения дерева проводника.
    ''' </summary>
    Public Property ShowTree As TItemShow Implements IEdmsExplorer.ShowTree
      Get
        Return GetRcwProperty("ShowTree")
      End Get
      Set
        SetRcwProperty("ShowTree", CInt(value))
      End Set
    End Property

    ''' <summary>
    ''' Получает содержимое дерева проводника.
    ''' </summary>
    ''' <returns>Объект IContents, состоящий из элементов типа IFolderInfo.</returns>
    Public Function GetTreeRootContents() As IContents(Of IFolderInfo) Implements IEdmsExplorer.GetTreeRootContents
      Dim rcwTreeRootContents = GetRcwProperty("TreeRootContents")
      Return New Contents(Of IFolderInfo)(rcwTreeRootContents, Scope)
    End Function

    ''' <summary>
    ''' Устанавливает содержимое дерева проводника.
    ''' </summary>
    ''' <param name="value">Значение.</param>
    Public Sub SetTreeRootContents(value As IFolderInfo) Implements IEdmsExplorer.SetTreeRootContents
      SetRcwProperty("TreeRootContents", DirectCast(value, IRcwProxy).RcwObject)
    End Sub

    ''' <summary>
    ''' Устанавливает содержимое дерева проводника.
    ''' </summary>
    ''' <param name="value">Значение.</param>
    Public Sub SetTreeRootContents(value As IContents(Of IFolderInfo)) Implements IEdmsExplorer.SetTreeRootContents
      SetRcwProperty("TreeRootContents", DirectCast(value, IRcwProxy).RcwObject)
    End Sub

    ''' <summary>
    ''' Показывает проводник.
    ''' </summary>
    Public Sub Show() Implements IEdmsExplorer.Show
      InvokeRcwInstanceMethod("Show")
    End Sub

    ''' <summary>
    ''' Конструктор.
    ''' </summary>
    ''' <param name="rcwIEdmsExplorer">СOM-объект IEdmsExplorer.</param>
    ''' <param name="scope">Область видимости.</param>
    Public Sub New(rcwIEdmsExplorer As Object, scope As IScope)
      MyBase.New(rcwIEdmsExplorer, scope)
    End Sub
  End Class
End NameSpace

File: src/IsbInterop/ServiceForms/Proxies/EdmsExplorer.cs
﻿using IsbInterop.Accessory;
using IsbInterop.Accessory.Proxies;
using IsbInterop.Collections;
using IsbInterop.Collections.Proxies;
using IsbInterop.DataTypes.Enumerable;
using IsbInterop.Folders;

namespace IsbInterop.ServiceForms.Proxies
{
  /// <summary>
  /// Обертка над IEdmsExplorer.
  /// </summary>
  public class EdmsExplorer : BaseIsbObject, IEdmsExplorer
  {
    /// <summary>
    /// Тип отображения дерева проводника.
    /// </summary>
    public TItemShow ShowTree
    {
      get { return (TItemShow)GetRcwProperty("ShowTree"); }
      set { SetRcwProperty("ShowTree", (int)value); }
    }

    /// <summary>
    /// Получает содержимое дерева проводника.
    /// </summary>
    /// <returns>Объект IContents, состоящий из элементов типа IFolderInfo.</returns>
    public IContents<IFolderInfo> GetTreeRootContents()
    {
      var rcwTreeRootContents = GetRcwProperty("TreeRootContents");
      return new Contents<IFolderInfo>(rcwTreeRootContents, Scope);
    }

    /// <summary>
    /// Устанавливает содержимое дерева проводника.
    /// </summary>
    /// <param name="value">Значение.</param>
    public void SetTreeRootContents(IFolderInfo value)
    {
      SetRcwProperty("TreeRootContents", ((IRcwProxy)value).RcwObject);
    }

    /// <summary>
    /// Устанавливает содержимое дерева проводника.
    /// </summary>
    /// <param name="value">Значение.</param>
    public void SetTreeRootContents(IContents<IFolderInfo> value)
    {
      SetRcwProperty("TreeRootContents", ((IRcwProxy)value).RcwObject);
    }

    /// <summary>
    /// Показывает проводник.
    /// </summary>
    public void Show()
    {
      InvokeRcwInstanceMethod("Show");
    }

    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="rcwIEdmsExplorer">СOM-объект IEdmsExplorer.</param>
    /// <param name="scope">Область видимости.</param>
    public EdmsExplorer(object rcwIEdmsExplorer, IScope scope) : base(rcwIEdmsExplorer, scope) { }
  }
}
